apps/aecore/lib/aecore/sync/sync.ex

```diff
@@ -544,7 +544,7 @@ defmodule Aecore.Sync.Sync do
 
   defp do_start_sync(peer_id, remote_header_hash) do
     case PeerConnection.get_header_by_hash(remote_header_hash, peer_id) do
-      {:ok, %{header: header}} ->
+      {:ok, %Header{} = header} ->
```

Comment:
I think the problem is somewhere in this file. Those are the only behaviour changes to sync.